controller/api/destination/server.go

```diff
@@ -277,11 +277,23 @@ func (s *server) getProfileByIP(
 
 	if svcID == nil {
 		// If the IP does not map to a service, check if it maps to a pod
-		pod, err := getPodByIP(s.k8sAPI, ip.String(), port, s.log)
+		var pod *corev1.Pod
+		targetIP := ip.String()
+		pod, err = getPodByHostIP(s.k8sAPI, ip.String(), port, s.log)
 		if err != nil {
 			return err
 		}
-		address, err := s.createAddress(pod, ip.String(), port)
+		if pod == nil {
+			pod, err = getPodByPodIP(s.k8sAPI, ip.String(), port, s.log)
+			if err != nil {
+				return err
+			}
+			if pod != nil {
+				targetIP = pod.Status.PodIP
+			}
+		}
```

Comment:
Would it make sense to check the pod network first? It seems like this will be the vastly more common case, so it bypasses doing a needless search on the host network. Or are we confident that that these searches are cheap enough that the ordering doesn't really matter?

Reply:
Good point. Both calls only use the informer's indices, so they should be cheap, but we can definitely save ourselves some cycles by doing it the other way around.